packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -436,6 +448,58 @@ function isConditionalTest(node: TSESTree.Node): boolean {
   return false;
 }
 
+function isMakeBoolean(
+  node: TSESTree.Node,
+  context: Readonly<TSESLint.RuleContext<MessageIds, Options>>,
+): boolean {
+  const parents = new Set<TSESTree.Node | null>([node]);
+  let current = node.parent;
+  while (current) {
+    parents.add(current);
+
+    if (
+      current.type === AST_NODE_TYPES.CallExpression &&
+      isBuiltInBooleanCall(current, context)
+    ) {
+      return true;
+    }
+
```

Comment:
So, the algorithm in this function definitely has some problems.... 

For example, the following code will be ignored by the new option
```ts
let a: string | true | undefined;
let b: string | boolean | undefined;

declare function f(x: unknown): unknown;

const x = Boolean(f(a || b));
```

However, I'm also noticing that the existing code works quite similarly.... 

https://github.com/typescript-eslint/typescript-eslint/blob/723d0a8ec655f89f0383ca8ed76158368b6d699b/packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts#L414-L449

So I'm wondering whether that has bugs too....

Without that context, I'd think we'd want to model this algorithm closer to what's in other rules, for example, no-floating-promises or no-extra-boolean-cast. Those rules recurse specifically on logical expressions, `??` expressions, ternaries, chain expressions, sequence expressions.

Reply:
upon checking, it appears that similar code is ignored when the ignoreConditionalTests option is true [playground](https://typescript-eslint.io/play/#ts=5.5.2&showAST=es&fileType=.ts&code=CYUwxgNghgTiAEEQBd5QFzwM7JgSwDsBzeAH3gIFcIIBuAKFElgTAHsCd4AjTHfYmQrU69RuGhx4AM0oEwyPBxkAKAB6Y5AawJsA7gQCUmgjv0EGeafBXSVUMuW6HD8AN4BfenoAWeJDZ2DqROLu5eQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6MAG1w2HL2jjokADRr1WddkiV5TbYgDC4pgBNK%2BSrekAVFPlQZ8cRAewAvv4AumpBAUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

```
declare let a: string | null;
declare const b: string | null;

declare function f(x: unknown): unknown;
if (f(a || b)) {}
while (f(a || b)) {}
```
so i think isConditionalTest function need to modified. Would it be a good idea to open a new issue for this problem?
_________________________________________________________________to modify algorithm, I looked at the `no-extra-boolean-cast` code you mentioned.

considering what you mentioned, it seems like you can improve it by referring to the algorithm of the function below, but I am wondering if my understanding is correct.

https://github.com/eslint/eslint/blob/main/lib/rules/no-extra-boolean-cast.js#L117-L167

Reply:
What fun, haha. Thanks for checking on this!

I think file no matter what, and it's up to you if you feel motivated and want to fix it in this PR or prefer to not worry about it in this work. 👍

Reply:
> considering what you mentioned, it seems like you can improve it by referring to the algorithm of the function below, but I am wondering if my understanding is correct.

Yeah, exactly, that's the pattern I'm suggesting to study 👍 (just the first thing that came to my mind, since [I have previously touched that code](https://github.com/eslint/eslint/pull/18222/files))